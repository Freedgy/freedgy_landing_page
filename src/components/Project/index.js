import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import Box from '@material-ui/core/Box';

const styles = theme => ({
    project: {
        "&&": {
            width: "100%",
            height: "100%",
            display: "flex",
            flexDirection: "column",
            alignItems: "center",
        }
    },
    content: {
        "&&": {
            marginTop: "2%",
            width: "80%",
            height: "100%",
            display: "flex",
            flexDirection: "column",
        }
    },
});

function Project({ classes, datas }) {
    return (
        <section id="project" className={classes.project}>
            <Typography variant="h2" align="center">
                {datas.title}
            </Typography>
            <Box className={classes.content}>
                <Typography variant="body1" align="justify">
                    {datas.description}
                </Typography>
            </Box>
        </section>
    );
}

Project.propTypes = {
    classes: PropTypes.object.isRequired,
    datas: PropTypes.object.isRequired,
};

export default withStyles(styles)(Project);