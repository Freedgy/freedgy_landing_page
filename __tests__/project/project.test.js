import React from "react"
import renderer from "react-test-renderer"

import Project from "../../src/components/Project"

describe("Project", () => {
    it("renders correctly", async () => {
        const datas = require("../../static/pages/pages.en.json");

        const tree = renderer
            .create(<Project datas={datas.content.project} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})