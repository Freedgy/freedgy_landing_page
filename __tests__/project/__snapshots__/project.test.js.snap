// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project renders correctly 1`] = `
<section
  className="Project-project-1"
  id="project"
>
  <h2
    className="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter"
  >
    Project
  </h2>
  <div
    className="MuiBox-root MuiBox-root-3 Project-content-2"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-alignJustify"
    >
      In the society of today, there is a lot of food waste. Furthermore, thousands of people struggle to earn enough money to feed themselves. We figured that most of the food waste was done before the citizens would get access to it: supermarkets throw away dozens of kilograms of consumable food every day. Today, there is no smart way for supermarkets to avoid wasting that food. That is when Freedgy come in. We will sell fridges to supermarkets and restaurants. Then they will be able to put the food they would otherwise throw away in those fridges.The fridges will be connected to our mobile application. On the mobile application, anyone like you and I will be able to donate food, and the supermarkets/restaurants in charge will simply put that donated food into the fridge.
    </p>
  </div>
</section>
`;
